src/server/game/AI/CoreAI/UnitAI.cpp

```diff
@@ -193,6 +193,35 @@ void UnitAI::DoCastAOE(uint32 spellId, bool triggered)
     me->CastSpell(nullptr, spellId, triggered);
 }
 
+BasicEvent* UnitAI::DoCastDelayed(Unit* victim, uint32 spellId, uint32 delay, bool triggered, bool aliveCheck)
+{
+    struct DelayedCast : public BasicEvent
+    {
+        DelayedCast(Unit *caster, Unit *target, uint32 spellId, bool triggered, bool aliveCheck) : BasicEvent(),
+            _caster(caster), _target(target), _spellId(spellId), _triggered(triggered), _aliveCheck(aliveCheck) { }
+
+        bool Execute(uint64, uint32)
+        {
+            if (_aliveCheck && _caster->IsAlive())
+                _caster->CastSpell(_target, _spellId, _triggered);
+            else
+                _caster->CastSpell(_target, _spellId, _triggered);
+            return true;
+        }
+
+        Unit *_caster;
+        Unit *_target;
```

Comment:
this leads to crashes
btw. since we have a taskscheduler i don't see any use for this function

Reply:
Because TaskScheduler wouldn't work if its not being updated from UpdateAI?

Reply:
`m_Events` still processes while dead. Why is this not sufficient?

Reply:
Sure this will suffice, but doesn't it feel hacky?
This would only allow us to cast spells with a certain delay, that aren't affected by death for example.

How about when we need to do more than just cast a delayed spell upon death.
I don't have a suggestion right now, but I don't see why we shouldn't be able to schedule after-death events properly, even though in 99% of the cases it's not relevant.

Reply:
You seem to be misunderstanding - I'm not agreeing with the approach of the PR.

I'm trying to figure out why (or whether) you think the current state of affairs - using custom `Event`s for the very few edge cases where this is relevant and no other master AI is around to handle it cleanly - is insufficient.

Reply:
I think using custom Events for special cases like this is fine.

Reply:
who won here ? is it still useful to have this event ?

Reply:
Doesn't seem like there's any convincing argument for it. Closing.